test/test_pytato_arraycontext.py

```diff
@@ -121,6 +121,72 @@ def twice(x):
     assert ran_callback
 
 
+@pytest.mark.parametrize("pass_allocator", ["auto_none", "auto_true", "auto_false",
+                                            "pass_buffer", "pass_svm",
+                                            "pass_buffer_pool", "pass_svm_pool"])
+def test_pytato_actx_allocator(actx_factory, pass_allocator):
+    base_actx = actx_factory()
+    alloc = None
+    use_memory_pool = None
+
+    if pass_allocator == "auto_none":
+        pass
+    elif pass_allocator == "auto_true":
+        use_memory_pool = True
+    elif pass_allocator == "auto_false":
+        use_memory_pool = False
+    elif pass_allocator == "pass_buffer":
+        from pyopencl.tools import ImmediateAllocator
+        alloc = ImmediateAllocator(base_actx.queue)
+    elif pass_allocator == "pass_svm":
+        from pyopencl.characterize import has_coarse_grain_buffer_svm
+        if not has_coarse_grain_buffer_svm(base_actx.queue.device):
+            pytest.skip("need SVM support for this test")
+        from pyopencl.tools import SVMAllocator
+        alloc = SVMAllocator(base_actx.queue.context, queue=base_actx.queue)
+    elif pass_allocator == "pass_buffer_pool":
+        from pyopencl.tools import ImmediateAllocator, MemoryPool
+        alloc = MemoryPool(ImmediateAllocator(base_actx.queue))
+    elif pass_allocator == "pass_svm_pool":
+        from pyopencl.characterize import has_coarse_grain_buffer_svm
+        if not has_coarse_grain_buffer_svm(base_actx.queue.device):
+            pytest.skip("need SVM support for this test")
+        from pyopencl.tools import SVMAllocator, SVMPool
+        alloc = SVMPool(SVMAllocator(base_actx.queue.context, queue=base_actx.queue))
+    else:
+        raise ValueError(f"unknown option {pass_allocator}")
+
+    actx = _PytatoPyOpenCLArrayContextForTests(base_actx.queue, allocator=alloc,
+                                               use_memory_pool=use_memory_pool)
+
+    def twice(x):
+        return 2 * x
+
+    f = actx.compile(twice)
+    res = actx.to_numpy(f(99))
+
+    assert res == 198
+
+    # Also test a case in which SVM is not available
+    if pass_allocator in ["auto_none", "auto_true", "auto_false"]:
+        def override_has_svm(dev):
+            return False
+
+        import pyopencl.characterize as cl_char
+
+        backup_has_coarse_grain_buffer_svm = cl_char.has_coarse_grain_buffer_svm
+        cl_char.has_coarse_grain_buffer_svm = override_has_svm
+
+        actx = _PytatoPyOpenCLArrayContextForTests(base_actx.queue, allocator=alloc,
+                                               use_memory_pool=use_memory_pool)
+        f = actx.compile(twice)
+        res = actx.to_numpy(f(99))
+
+        assert res == 198
+
+        cl_char.has_coarse_grain_buffer_svm = backup_has_coarse_grain_buffer_svm
```

Comment:
https://docs.python.org/3/library/unittest.mock.html is better for this. Suppose something crashes in this test, it'll leave the module in a weird state. (Where as `mock` handles exceptions.)

Reply:
Mock is great ;-). Done in [9585727](https://github.com/inducer/arraycontext/pull/197/commits/9585727e8087ca5e14824c8aee29ba8f3e147d87)

Reply:
Thanks. LGTM!